### PR TITLE
silo: Cleanup the HDF5 dependency specs

### PR DESCRIPTION
**Pull request by chuckatkins**

### CODE REVIEW

**Comment by sethrj** [comment]
(See #29132 for details of why I added the extra constraints.)

File: var/spack/repos/builtin/packages/silo/package.py [hunk 1]
```diff
@@ -44,8 +44,7 @@ class Silo(AutotoolsPackage):
     depends_on('automake', type='build', when='+shared')
     depends_on('libtool', type='build', when='+shared')
     depends_on('mpi', when='+mpi')
-    depends_on('hdf5', when='+hdf5')
-    depends_on('hdf5 api=v110', when='@:4.10 +hdf5 ^hdf5@1.12:')
+    depends_on('hdf5@1.8:', when='+hdf5')
```
Comment (kwryankrattiger) [inline var/spack/repos/builtin/packages/silo/package.py:47]: This should be constrained to `@1.10` right? hdf5@1.8 doesn't support the macros for the 1.10 API.

**Comment by kwryankrattiger** [comment]
> (See #29132 for details of why I added the extra constraints.)

Using the `api` variant is kind of problematic. The way this is written now, it will use HDF5s macros to resolve the required API for the project without messing with other projects that need the API version of the hdf5 version that is built.

**Review by kwryankrattiger — APPROVED** [review]
These look good to me @alalazo

**Pull request merged**